### ISSUE TITLE
Callbacks cannot reset.

### ISSUE THREAD

**Issue by muhammadAgfian96** [ultralytics/ultralytics#3063]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Integrations, Other

### Bug

i want to integrate clearml. since yolov8 already integrate with clearml, i want to make my own custom and override the callbacks from default clearml. but i tried to reset callbacks it seems the callbacks still there and mix with my callback functions. 

```
model_yolo = YOLO(model=args_task["model_name"] + ".pt")

# add my custom callbacks
model_yolo._reset_callbacks()
for event, func in callbacks.items():
    model_yolo.clear_callback(event)
    model_yolo.add_callback(event, func)
```
callbacks is ok here and expected implement my callbacks, but after this step:
```
model_yolo.train(data=data_yaml_file, **args)
```
its calling callbacks from default clearml in yolov8 and mix with my callbacks

is it expected or i do something wrong? 
TIA.

### Environment

--index-url https://download.pytorch.org/whl/cu117
torch==2.0.0+cu117
torchvision==0.15.1+cu117

ultralytics==8.0.112
clearml==1.11.0

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by muhammadAgfian96** [comment]
can we disable callbacks.add_integration_callbacks(self)?

https://github.com/ultralytics/ultralytics/blob/29c954a1385815e522f0d94d7144814cc4e6da42/ultralytics/yolo/engine/trainer.py#L147-L148

or can we choose to disable default callbacks (comet, clearml, dvc)

**Comment by glenn-jocher** [comment]
Hi @muhammadAgfian96 👋,

Regarding your question, YOLOv8 callbacks are intended to be used as hooks during the training process, but it is possible to disable some of them with specific environment flags, as you pointed out. However, the purpose of these callbacks is to support integrations with popular ML platforms and tools like Comet, Clearml, and DVC, so disabling them may prevent you from getting valuable data about code and model performance during training.

If you have any additional requirements for training and logging models, we recommend either using the [UltraLytics](https://ultralytics.com/) API or extending YOLOv8 with your own custom code that meets your specific requirements. If you would like to submit a pull request with your changes, please ensure that your changes do not affect the existing functionality of YOLOv8.

Let us know if this information helps, and feel free to reach out if you have any further questions or concerns! 😃

**Comment by muhammadAgfian96** [comment]
Hi @glenn-jocher 

Thanks for engaging with our community and sharing your insights. Your response is greatly appreciated.

Currently, I'm encountering a challenge when integrating ClearML with our project. To elaborate, I've developed a set of custom callbacks using ClearML as follows:

```
# main_train.py
from my_callbacks import callbacks
...
model_yolo = YOLO(model=args_task["model_name"] + ".pt")

for event, func in callbacks.items():
    model_yolo.clear_callback(event)
    model_yolo.add_callback(event, func)

model_yolo.train(data=data_yaml_file, **args)
```

Initially, I presumed this approach would override the default ClearML callbacks, but it appears the result is a fusion of both my custom callbacks and the default ones. As a result, we end up with an amalgamation of two callback sets.

To remedy this, I'm considering submitting a PR that would disable the default ClearML callbacks. This way, users of ClearML will have the flexibility to implement their custom ClearML callbacks without the interference of default ones.

Your thoughts on this proposition would be greatly valued.

**Comment by glenn-jocher** [comment]
@muhammadAgfian96 great news 😃! Your original issue may now be resolved ✅ in PR #3790 and published in a new `ultralytics 8.0.141` release. This PR adds a SettingsManager class that allows for direct viewing and modifying of Ultralytics settings:

<img width="955" alt="Screenshot 2023-07-23 at 16 11 23" src="https://github.com/ultralytics/ultralytics/assets/26833433/1427f5bd-52b6-445a-ab5b-e4c0eeccc10b">

For details see https://docs.ultralytics.com/quickstart

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by muhammadAgfian96** [comment]
it works really simple! Thanks @glenn-jocher !

**Comment by glenn-jocher** [comment]
You're welcome @muhammadAgfian96! I'm glad to hear it worked well for you. If you run into any more issues or need further assistance, feel free to reach out. Happy coding!
